src/sentry/tasks/post_process.py

```diff
@@ -468,10 +481,38 @@ def post_process_group(
                 return
             # Issue platform events don't use `event_processing_store`. Fetch from eventstore
             # instead.
-            event = eventstore.get_event_by_id(
-                project_id, occurrence.event_id, group_id=group_id, skip_transaction_groupevent=True
+
+            def get_event_with_retry(
+                getter_func: Callable[[], Optional[Event]],
+                retry_exceptions: Sequence[Type[Exception]] = (),
+                attempts=3,
+            ) -> Optional[Event]:
+                attempt = 1
+                delay_exponent = 0
+                event_or_none = None
+                while event_or_none is None or attempt <= attempts:
+                    time.sleep(pow(2, delay_exponent))
+                    try:
+                        event_or_none = getter_func()
+                    except retry_exceptions:
+                        pass
+                    delay_exponent += 1
+                    attempt += 1
+                return event_or_none
+
+            event = get_event_with_retry(
+                lambda: eventstore.get_event_by_id(
+                    project_id,
+                    occurrence.event_id,
+                    group_id=group_id,
+                    skip_transaction_groupevent=True,
+                ),
+                [ServiceUnavailable],
             )
 
+        if event is None:
+            return
```

Comment:
Probably worth logging an error here or at least a metric

Reply:
Right, I modified the PR a bit so it would just re-raise the exception `EventLookupError` if the retry attempts are exhausted.